src/control/lib/hardware/topology.go

```diff
@@ -47,6 +47,27 @@ func (t *Topology) AllDevices() map[string]*PCIDevice {
 	return devsByName
 }
 
+// NumNUMANodes gets the number of NUMA nodes in the system topology.
+func (t *Topology) NumNUMANodes() int {
+	if t == nil {
+		return 0
+	}
+	return len(t.NUMANodes)
+}
+
+// NumCoresPerNUMA gets the number of cores per NUMA node.
+func (t *Topology) NumCoresPerNUMA() int {
```

Comment:
I'm not sure that the assumption here is correct -- particularly when run on "weird" (e.g. with on-die GPUs, etc) systems, I think it's possible to have non-homogenous core counts per node. Whether or not this is relevant to anything, especially now, I don't know... But I wanted to point it out.

Reply:
I was thinking about this, too. Currently we have a drpc request that incorporates this assumption, that uses this call directly. I think we'll need to update that message, and then this can be removed.

Reply:
Yeah, after I wrote this I realized that the guts had come from netdetect. I'm fine with removing it later.